### PR TITLE
[CT-1050] DeliverTx state change reset for subaccount updates

### PR DESCRIPTION
**Pull request by dydxwill**
### Changelist
[Describe or list the changes made in this PR]

### Test Plan
[Describe how this PR was tested (if applicable)]

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

## Summary by CodeRabbit

- **New Features**
  - Introduced a new streaming mechanism for managing subaccount updates.
  - Added support for real-time updates specific to subaccounts, enhancing responsiveness and data accuracy.
  - New metrics for tracking gRPC operations related to subaccount snapshots and updates.

- **Bug Fixes**
  - Improved handling of subscription management for subaccounts, ensuring timely updates.

- **Documentation**
  - Updated documentation to reflect changes in subaccount management and streaming functionalities.

- **Chores**
  - Streamlined CI/CD workflows by reducing redundant job definitions and enhancing maintainability.

### CODE REVIEW

File: protocol/x/clob/keeper/process_operations.go [hunk 1]
```diff
@@ -58,6 +73,7 @@ func (k Keeper) ProcessProposerOperations(
 	}
 
 	// If grpc streams are on, send absolute fill amounts from local + proposed opqueue to the grpc stream.
+	// Also send subaccount snapshots for impacted subaccounts.
```
Comment (jonfung-dydx) [inline protocol/x/clob/keeper/process_operations.go:76]:
```suggestion
	// Also send subaccount snapshots for impacted subaccounts.
	// An impacted subaccount is defined as:
	// - A subaccount that was involved in any match in the local opqueue.
	//   Only matches generate subaccount updates.
```

File: protocol/x/clob/keeper/process_operations.go [hunk 2]
```diff
@@ -75,6 +91,21 @@ func (k Keeper) ProcessProposerOperations(
 			allUpdates.Append(orderbookUpdate)
 		}
 		k.SendOrderbookUpdates(ctx, allUpdates)
+
+		// send subaccount snapshots
+		subaccountIdsFromProposed := fetchSubaccountIdsInvolvedInOpQueue(
```
Comment (jonfung-dydx) [inline protocol/x/clob/keeper/process_operations.go:96]: super meganit, up to you: combine the two loops where we iterate through the opqueue?

File: protocol/x/subaccounts/keeper/subaccount.go [hunk 3]
```diff
@@ -132,6 +132,34 @@ func (k Keeper) GetSubaccount(
...
+			PerpetualId: pp.PerpetualId,
+			Quantums:    pp.Quantums.BigInt().Uint64(),
+		}
+	}
+
+	return types.StreamSubaccountUpdate{
+		SubaccountId:              &id,
+		UpdatedAssetPositions:     assetPositions,
+		UpdatedPerpetualPositions: perpetualPositions,
+		Snapshot:                  true,
```
Comment (jonfung-dydx) [inline protocol/x/subaccounts/keeper/subaccount.go:159]: I don't think this is a snapshot

**Review by jonfung-dydx — APPROVED** [review]

File: protocol/x/clob/keeper/process_operations.go (hunk 2 above)
Comment (dydxwill) [reply protocol/x/clob/keeper/process_operations.go:99]: only sending local for now

File: protocol/x/subaccounts/keeper/subaccount.go (hunk 3 above)
Comment (dydxwill) [reply protocol/x/subaccounts/keeper/subaccount.go:159]: Why isn't it? We are getting the subaccount from state which should have all asset/perp positions?

**Pull request merged**